### PR TITLE
:sparkles: Wire up analysis progress and display rule IDs

### PR DESCRIPTION
**Pull request by tsanders-rh**
## Summary

Improves analysis progress reporting to display actual rule IDs being processed and fixes garbled JSON output in the OUTPUT panel.

### Changes:

1. **Display Rule IDs in Progress Messages**
   - Shows the actual rule ID being processed (e.g., "Processing rule 49/56: java-transaction-00001")
   - Abbreviated message in VS Code notification: "Processing rule 49/56"
   - Detailed message with rule ID in Analysis View webview
   - Maps progress percentages appropriately (0% → 10% → 20% → 90% → 100%)
   - **Backward compatible**: Gracefully falls back to showing just "Processing rule 49/56" if analyzer doesn't send rule IDs

2. **Fix Garbled JSON in OUTPUT Panel**
   - Filters out non-progress JSON from stderr to prevent clutter
   - Only logs actual error messages (non-JSON lines) to OUTPUT panel
   - Fixes issue where raw JSON appeared when laptop wakes from sleep
   - **No dependency**: This fix works independently of analyzer-rpc version

3. **Comprehensive Testing**
   - Added 15 unit tests for `ProgressParser` with 100% coverage
   - Tests cover progress parsing, JSON filtering, buffer handling, and edge cases
   - All tests passing ✓

4. **Documentation**
   - Added JSDoc comments to explain filtering behavior
   - Documented dual UI update strategy (notification vs webview)
   - Clarified that `event.message` contains different data per stage

## Dependency (Optional Enhancement)

This PR is **backward compatible** and will work with the current analyzer-rpc binary. However, to display rule IDs, the `kai-analyzer-rpc` needs to include the rule ID in progress events:

```json
{
  "timestamp": "2024-01-01T00:00:00Z",
  "stage": "rule_execution",
  "current": 49,
  "total": 56,
  "message": "java-transaction-00001",  // ← Rule ID here
  "metadata": {
    "rule_id": "java-transaction-00001"  // ← Or here
  }
}
```

**Without analyzer changes**: Shows "Processing rule 49/56" (still an improvement over current state)  
**With analyzer changes**: Shows "Processing rule 49/56: java-transaction-00001"

**Related kai PR**: https://github.com/konveyor/kai/pull/895

### Before:
- Progress message: "Processing rules: 49/56 (87.5%)"
- OUTPUT panel cluttered with raw JSON after sleep/wake
- No visibility into which rule is being processed

### After:
- **Notification**: "Processing rule 49/56" (clean, abbreviated)
- **Webview**: "Processing rule 49/56: java-transaction-00001" (detailed with rule ID, if available)
- **OUTPUT panel**: Only actual error messages, no JSON clutter ✓

### Automated Testing:
```bash
cd vscode/core
npm test
```

**Results**: 15/15 ProgressParser tests passing ✓

### Manual Testing Checklist:
- [x] Progress messages display correctly in VS Code notification
- [x] Progress messages display correctly in Analysis View webview
- [x] OUTPUT panel only shows error messages (no JSON)
- [x] Works with current analyzer-rpc binary (backward compatible)
- [ ] Rule IDs display when analyzer-rpc includes them (pending konveyor/kai#895)
- [ ] Sleep/wake scenario doesn't produce garbled output

## Commits

1. **feat**: Display rule ID in analysis progress messages (4e4a5e4)
2. **fix**: Prevent raw progress JSON from appearing in OUTPUT panel (de10ee3)
3. **fix**: Skip all JSON output to prevent clutter in OUTPUT panel (c2d14ed)
4. **docs**: Add JSDoc comments to progress parser and callback logic (c519ee4)
5. **test**: Add comprehensive unit tests for ProgressParser (11c140c)

## Files Changed

- `vscode/core/src/client/progressParser.ts` - Enhanced parser with JSON filtering + JSDoc
- `vscode/core/src/client/analyzerClient.ts` - Updated progress callback logic + JSDoc
- `vscode/core/src/client/__tests__/progressParser.test.ts` - **NEW** 15 comprehensive tests

## Checklist

- [x] Code follows project style guidelines
- [x] All tests passing (15 new tests, 34 total)
- [x] JSDoc comments added
- [x] Commits signed off with DCO
- [x] Pre-commit hooks passing (eslint)
- [x] Backward compatible with current analyzer-rpc
- [ ] Tested with analyzer-rpc that sends rule IDs (waiting for konveyor/kai#895 to merge)
- [ ] Ready to mark as ready for review (currently draft)

## Why Draft?

This PR is marked as draft pending konveyor/kai#895 which adds rule ID reporting to the analyzer. The code is backward compatible and CI should pass, but marking as draft until both PRs can be tested together for the full feature.

## Summary by CodeRabbit

* **New Features**
  * Real-time analysis progress with numeric percent and descriptive messages shown in UI.
  * Analysis page displays a dynamic progress indicator and message during runs.
  * Progress reporting integrated into the analysis pipeline for live updates and state visibility (including new progress fields).

* **Bug Fixes**
  * Progress now resets when analysis completes and non-progress output is separated from progress events.

* **Tests**
  * Added comprehensive tests for parsing progress/event streams and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: vscode/core/src/client/progressParser.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+   * @param line - A single line from stderr to parse
+   */
+  private parseLine(line: string): void {
+    try {
+      const obj = JSON.parse(line);
+      if (this.isProgressEvent(obj)) {
+        this.callback(obj);
+      }
+      // Skip other JSON - don't log it to prevent clutter in OUTPUT panel
+    } catch (err) {
```
Comment (djzager) [inline vscode/core/src/client/progressParser.ts]: nitpick.

```suggestion
    } catch {
```

**Review by djzager — DISMISSED** [review]
Non-blocking nitpick. Otherwise lgtm.
